client/lib/perfmon/index.js

```diff
@@ -0,0 +1,173 @@
+	if ( placeholdersVisibleStart && ( visibleCount === 0 || trigger === 'navigate' ) ) {
+		// tell tracks to record duration
+		var duration = Date.now() - placeholdersVisibleStart;
+		debug(`Recording placeholder wait. Duration: ${duration}, Trigger: ${trigger}`);
+		analytics.timing.record( `placeholder-wait`, duration, trigger );
+		placeholdersVisibleStart = visibleCount === 0 ? null : Date.now();
+	}
+
+	// if we can see placeholders, placeholdersVisibleStart is falsy, start the clock
+	if ( visibleCount > 0 && !placeholdersVisibleStart ) {
+		placeholdersVisibleStart = Date.now(); // TODO: performance.now()?
+	}
+
+	// if there are placeholders hanging around, print some useful stats
+	if ( activePlaceholders.length > 0 ) {
+		debug("Active placeholders: "+activePlaceholders.length);
+		debug("Visible in viewport: "+visibleCount);
+	}
+}
+
+function isPlaceholder( node ) {
+	var className = node.className;
+	return className && className.indexOf && PLACEHOLDER_CLASSES.some( function( clazz ) { 
+		return (className.indexOf( clazz ) >= 0); 
+	} );
+}
+
+// adapted from http://stackoverflow.com/questions/123999/how-to-tell-if-a-dom-element-is-visible-in-the-current-viewport/7557433#7557433
+function isElementVisibleInViewport( node ) {
+	var rect = node.getBoundingClientRect();
+
+	// discard if width or height are zero
+	if ( (rect.top - rect.bottom) === 0 || (rect.right - rect.left) === 0) {
+		return false;
+	}
+
+	var windowHeight = (window.innerHeight || document.documentElement.clientHeight);
+	var windowWidth = (window.innerWidth || document.documentElement.clientWidth);
+
+	// if top or bottom are inside window, AND left or right edge are inside window, then
+	// the element is at least partially visible
+	return (
+		( ( rect.top >= 0 && rect.top <= windowHeight ) || ( rect.bottom >= 0 && rect.bottom <= windowHeight ) ) &&
+		( ( rect.left >= 0 && rect.left <= windowWidth ) || ( rect.right >= 0 && rect.right <= windowWidth  ) )
+	);
+}
+
+function recordPlaceholders( mutation ) { 
+	var nodes = [];
+
+	if ( mutation.attributeName === 'class' ) { // mutation.type === 'attributes' is redundant
+		nodes = [mutation.target];
+	} else if ( mutation.type === 'childList' && mutation.addedNodes.length > 0 ) {
+		nodes = mutation.addedNodes;
+	} else {
+		return;
+	}
+
+	each( nodes, function( node ) {
+
+		if ( isPlaceholder( node ) ) {
+			recordPlaceholderNode( node );
+		}
+
+		// we need to find matching children because the mutation observer
+		// only fires for the top element of an added subtree
+		if ( node.querySelectorAll ) {
+			// funky syntax because NodeList walks like an array but doesn't quack like one
+			each( node.querySelectorAll(PLACEHOLDER_MATCHER), recordPlaceholderNode );
+		}
+	} );
+}
+
+function recordPlaceholderNode( node ) {
+	if ( activePlaceholders.indexOf( node ) >= 0 ) {
+		// no-op
+	} else {
+		activePlaceholders.push(node);
+	}
+}
+
+module.exports = function() {
```

Comment:
It looks like you have embraced the ES6 syntax in the rest of this file, how about an `export`?

```js
export default function() {
	…
}
```